generator/template/Emoji.java

```diff
@@ -44,27 +62,52 @@
   }
 
   @NonNull @Override public Drawable getDrawable(final Context context) {
-    if (sheet == null) {
-      synchronized (LOCK) {
-        if (sheet == null) {
-          sheet = BitmapFactory.decodeResource(context.getResources(), R.drawable.emoji_<%= package %>_sheet);
-        }
-      }
+    final CacheKey key = new CacheKey(x, y);
+    final Bitmap bitmap = BITMAP_CACHE.get(key);
+    if (bitmap != null) {
+      return new BitmapDrawable(context.getResources(), bitmap);
     }
-
-    final Bitmap cut = Bitmap.createBitmap(sheet, x * 66, y * 66, 64, 64);
-
+    final Bitmap strip = loadStrip(context);
+    final Bitmap cut = Bitmap.createBitmap(strip, 1,
+        y * SPRITE_SIZE_INC_BORDER + 1, SPRITE_SIZE, SPRITE_SIZE);
+    BITMAP_CACHE.put(key, cut);
     return new BitmapDrawable(context.getResources(), cut);
   }
 
-  @Override public void destroy() {
-    if (sheet != null) {
+  private Bitmap loadStrip(final Context context) {
+    Bitmap strip = (Bitmap) STRIP_REFS[x].get();
+    if (strip == null) {
       synchronized (LOCK) {
-        if (sheet != null) {
-          sheet.recycle();
-          sheet = null;
+        strip = (Bitmap) STRIP_REFS[x].get();
+        if (strip == null) {
+          final Resources resources = context.getResources();
+          final int resId = resources.getIdentifier("emoji_<%= package %>_sheet_" + x,
+              "drawable", context.getPackageName());
+          strip = BitmapFactory.decodeResource(resources, resId);
+          STRIP_REFS[x] = new SoftReference<>(strip);
         }
       }
     }
+    return strip;
+  }
+
+  private static final class CacheKey {
```

Comment:
Yes sounds good!

Reply:
Moved to main module, in com.vanniktech.emoji.emoji. I had to make it public rather than package-protected because the various template-based Emoji classes are in different packages.